ballerina/client.bal

```diff
@@ -357,760 +358,425 @@ public isolated client class Client {
-                                            @display {label: "Fields to Retrieve"}
-                                          string... fields)
-                                            returns @display {label: "Result"} json|Error {
-        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sobject, id]);
-        if fields.length() > 0 {
-            path = path.concat(utils:appendQueryParams(fields));
+    # + userId - User ID
+    # + return - `byte[]` if successful or else `error`
+    isolated remote function resetPassword(string userId) returns byte[]|error {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, USER, userId, PASSWORD]);
+        http:Response response = check self.salesforceClient->delete(path);
+        if response.statusCode == 200 {
+            json payload = check response.getJsonPayload();
+            map<json> payloadMap = check payload.ensureType();
+            string password = check payloadMap.NewPassword;
+            return password.toBytes();
+        } else {
+            return error("Error occurred while resetting the password");
         }
-        json response = check self.get(path);
-        return response;
     }
 
-    # Gets an object record by external ID.
+    # Change user password
     #
-    # + sobject - SObject name 
-    # + extIdField - External ID field name 
-    # + extId - External ID value 
-    # + fields - Fields to retrieve 
-    # + return - JSON result if successful or else `sfdc:Error`
-    #
-    # # Deprecated
-    # This function is deprecated as it returns a json response.
-    # Use the new and improved `getByExternalId(string sObject, string extIdField, string extId)` function instead.
-    @display {label: "Get Record by External ID"}
-    @deprecated
-    isolated remote function getRecordByExtId(@display {label: "SObject Name"} string sobject,
-                                            @display {label: "External ID Field Name"} string extIdField,
-                                            @display {label: "External ID"} string extId,
-                                            @display {label: "Fields to Retrieve"}
-                                              string... fields)
-                                            returns @display {label: "Result"} json|Error {
-        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sobject, extIdField, extId]);
-        if fields.length() > 0 {
-            path = path.concat(utils:appendQueryParams(fields));
-        }
-        json response = check self.get(path);
-        return response;
+    # + userId - User ID
+    # + newPassword - New user password as a byte[]
+    # + return - `()` if successful or else `error`
+    isolated remote function changePassword(string userId, byte[] newPassword) returns error? {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, USER, userId, PASSWORD]);
+        record{} payload = {"NewPassword" : check string:fromBytes(newPassword)};
```

Comment:
Can we pass byte[] value directly?

Reply:
it gives unknown exception from salesforce